docs/en/stack/security/troubleshooting.asciidoc

```diff
@@ -392,6 +392,123 @@ Kerberos/SPNEGO debug logging on JVM, add following JVM system properties:
+Cannot find metadata for entity [your:entity.id] in [metadata.xml]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+unable to authenticate user [<unauthenticated-saml-user>]
+for action [cluster:admin/xpack/security/saml/authenticate]
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+Authentication to realm my-saml-realm failed -
+Provided SAML response is not valid for realm saml/my-saml-realm
+(Caused by ElasticsearchSecurityException[SAML Response is not a 'success' response:
+Code=urn:oasis:names:tc:SAML:2.0:status:AuthnFailed Message=null Detail=null])
+-----------------------------------------------
+--
+
+. Authentication in Kibana fails and the following error is printed in the
+Elasticsearch logs:
++
+--
+[source, shell]
+-----------------------------------------------
+The XML Signature of this SAML message cannot be validated. Please verify that the saml
+realm uses the correct SAMLmetadata file/URL for this Identity Provider
+
+-----------------------------------------------
+--
+
+. Users are unable to login with a local username/password in Kibana because SAML is
+enabled.
+
+
+*Resolution:*
+
+. This can be resolved by correcting your `sp.acs` setting in your {es} SAML
+realm configuration or by using the `xpack.security.public.*` settings in `kibana.yml`.
+{kib} either uses the `server.host` and `server.port` or the `xpack.security.public.*`
+settings to construct the SAML ACS URL and this URL needs to match exactly the URL that
+is configured in the SAML realm configuration.
+
+NOTE: The two URLs are compared as case-sensitive strings and not as canonicalized URLs.
+
+. We could not find the metadata for the SAML Entity ID `your:entity.id` in the metadata
+file (`metadata.xml`) that has been configured. Either the value of `idp.entity_id` setting
+in the {es} SAML realm configuration is wrong/misspelled, or the wrong metadata file has been
+configured in `idp.metadata.path`.
+
+. This error indicates that {es} failed to process the incoming SAML authentication message.
+Since the message can't be processed, {es} is not aware of who the to-be authenticated
+user is and the `<unauthenticated-saml-user>` placeholder is used instead. To diagnose the
+_actual_ problem one needs to check the {es} logs for further details.
+
+. This means that the SAML Identity Provider failed to authenticate the user and sent a
+SAML Response to the Service Provider (Elastic Stack) indicating this failure. The `Code`
+can be a different error identifier than the one above and might offer an indication about
+the cause of the failure, but `Message` and `Detail` will be `null`. In order to further
```

Comment:
> `Message` and `Detail` will be `null`.

They might not be (but usually are).
If they're `null`, then that means we have no additional information, but it's possible for the IdP to populate them.

I think we want to say something more like:
- If the IdP provided details about the cause of the error, then this will be included in the `Message` or `Detail` attributes of the log message.
- If these attributes are `null`, then the IdP did not provide additional details and you will need to consult the logs and/or documentation of the Identity Provider.